### PR TITLE
Add a brief note to the README that the C++ server is deprecated

### PR DESCRIPTION
**Pull request by Martin2112**
And a longer explanation, based on a mail to the list from Pierre.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,22 @@ certificate-transparency: Auditing for TLS certificates
...
+
+[CT Personality](https://github.com/google/certificate-transparency-go)
+[Generic Backend](https://github.com/google/trillian)
+
+For more information on why we recommend this see the
+[Deprecation Notes](DEPRECATED.md)
+
+N.B. This notice refers to the servers only, not the other client code in this
+repository.
+
```
Comment (daviddrysdale) [inline README.md]: Could we move this section below the Introduction, and add a cross-reference to it from the first bullet of the introduction?
```
...The main areas covered are:
 - (**Deprecated**, see [below](#c++-log-server-deprecation-notice)). An open-source, distributed, implementation of a CT Log server, also including:
     - An implementation of a read-only ["mirror" server](docs/MirrorLog.md)
       that mimics a remote Log.
     - Ancillary tools needed for managing and maintaining the Log.
```

Might also be worth shuffling the bullets in the Introduction so the deprecated server is listed last.

File: DEPRECATED.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+The CT log server implementation in this repository is no longer under
+active development. We recommend that new deployments use the new Go
+based server, which can handle much bigger trees:
+
+[CT Personality](https://github.com/google/certificate-transparency-go)
+[Trillian Log Backend](https://github.com/google/trillian)
+
+The C++ log server implementation in this repository keeps its Merkle tree in
+memory, which means two things: before it can handle requests, it has to load
```
Comment (daviddrysdale) [inline DEPRECATED.md]: Maybe break the two things into bullets?

File: DEPRECATED.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+all the leaf hashes, which adds a significant startup delay (some log operators
+have seen startup time upwards of an hour), and also, that there's upper limits
+on the number of entries a log server can handle, related to the memory usage.
+
+Trillian provides a storage-based general (based on blobs rather than X.509 and
+such structures) Merkle tree platform that is much more scalable. A given
+request might have higher latency, due to having to perform lookups into the
+storage system, but fairly constant, and in exchange, the startup time is
+constant, and the number of entries a log server can contain is vastly higher
+(based on available disk storage rather than memory).
```
Comment (daviddrysdale) [inline DEPRECATED.md]: Could we split this sentence up into smaller chunks with fewer clauses?

File: DEPRECATED.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+CTFE (which is in the certificate-transparency-go repository) then simply
+implements the CT API on top of Trillian.
+
+Also, while the "classic" Google CT logs (such as Rocketeer, Pilot, Icarus, etc)
+are written in C++, they share very little code with the open source C++ log
+server (the low level Merkle tree code, and some serialisation code), but the
+newer Google CT logs (such as the Argon and Xenon logs) are Trillian based to a
+much greater degree, sharing something like over 80% of the code with the open
+source version (there are some interfaces for storage and a few other things
+that we have different implementations).
```
Comment (daviddrysdale) [inline DEPRECATED.md]: A one-sentence paragraph?  Have you been watching "Yes, Minister" again?  :-)

Comment (Martin2112) [reply DEPRECATED.md]: Copied from Pierre's email. As was the above.

File: README.md (hunk 1 above)
Comment (Martin2112) [reply README.md]: Done.

File: DEPRECATED.md (hunk 2 above)
Comment (Martin2112) [reply DEPRECATED.md]: Done.

File: DEPRECATED.md (hunk 3 above)
Comment (Martin2112) [reply DEPRECATED.md]: Done.

File: DEPRECATED.md (hunk 4 above)
Comment (Martin2112) [reply DEPRECATED.md]: Done.

**Pull request merged**